src/fmeval/transforms/common.py

```diff
@@ -14,18 +14,40 @@ class GeneratePrompt(Transform):
     applying the same prompt template to each input.
     """
 
-    def __init__(self, input_keys: List[str], output_keys: List[str], prompt_template: str):
+    def __init__(
+        self,
+        input_keys: List[str],
+        output_keys: List[str],
+        prompt_template: str,
+        placeholder_to_record_key: Optional[Dict[str, str]] = None,
+    ):
         """GeneratePrompt initializer.
 
-        :param input_keys: The keys corresponding to the text that will be used to create prompts.
-            If multiple input keys are provided, a prompt will be constructed from each input, but
-            the created prompts will all utilize the same prompt template.
-        :param output_keys: The keys corresponding to the prompts that get added by this Transform.
-        :param prompt_template: The template used to construct the prompt.
-            Example: "Summarize the following text: $model_input".
+                :param input_keys: The keys corresponding to the text that will be used to create prompts.
+                    If multiple input keys are provided, a prompt will be constructed from each input, but
+                    the created prompts will all utilize the same prompt template.
+                :param output_keys: The keys corresponding to the prompts that get added by this Transform.
+                :param prompt_template: The template used to construct the prompt.
+                    Example: "Summarize the following text: $model_input".
+                :param placeholder_to_record_key: The placeholders and the corresponding record keys dict.
+                    Note that when using `placeholder_to_record_key`, having input keys or more than one output key
+                    doesn't make much sense, as all composed prompts will be identical.
+                    Example:
+                        Inputs:
+                            prompt_template = "Summarize $x and $y"
+                            input_keys = []
+                            output_keys = ["my_prompt"]
+                            placeholder_to_record_key = {"x": "statement_1", "y": "statement_2"}
+                            record = {"statement_1": "some long text", "statement_2": "some other long text"}
+                        Output record (only new keys and values are shown):
+                            {"my_prompt": "Summarize some long text and some other long text"}
+
+        Output record (only new keys and values are shown):
+        {"my_prompt": "Summarize some long text and some other long text"}
         """
-        super().__init__(input_keys, output_keys, prompt_template)
+        super().__init__(input_keys, output_keys, prompt_template, placeholder_to_record_key)
         self.register_input_output_keys(input_keys, output_keys)
```

Comment:
Shoot, I didn't catch this before the PR got merged, but we will need to update the call to `self.register_input_output_keys` to handle the case where `placeholder_to_record_key` is non-null.

```
input_keys_to_register = list(placeholder_to_record_keys.values()) if placeholder_to_record_key else input_keys
self.register_input_output_keys(input_keys_to_register, output_keys)
```